crates/ruma-events/src/room/name.rs

```diff
@@ -11,51 +13,71 @@ pub type NameEvent = StateEvent<NameEventContent>;
 /// The payload for `NameEvent`.
 #[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
 #[ruma_event(type = "m.room.name", kind = State)]
+#[non_exhaustive]
 pub struct NameEventContent {
-    /// The name of the room. This MUST NOT exceed 255 bytes.
-    #[serde(default, deserialize_with = "room_name")]
-    pub(crate) name: Option<String>,
+    /// The name of the room.
+    #[serde(default, deserialize_with = "ruma_serde::empty_string_as_none")]
+    pub name: Option<RoomName>,
 }
 
 impl NameEventContent {
     /// Create a new `NameEventContent` with the given name.
-    ///
-    /// # Errors
-    ///
-    /// `InvalidInput` will be returned if the name is more than 255 bytes.
-    pub fn new(name: String) -> Result<Self, InvalidInput> {
-        match name.len() {
-            0 => Ok(Self { name: None }),
-            1..=255 => Ok(Self { name: Some(name) }),
-            _ => Err(InvalidInput("a room name cannot be more than 255 bytes".into())),
-        }
+    pub fn new(name: Option<RoomName>) -> Self {
+        Self { name }
     }
 
     /// The name of the room, if any.
-    pub fn name(&self) -> Option<&str> {
-        self.name.as_deref()
+    pub fn name(&self) -> &Option<RoomName> {
+        &self.name
     }
 }
 
-fn room_name<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
-where
-    D: serde::de::Deserializer<'de>,
-{
-    use serde::de::Error;
-
-    // this handles the null case and the empty string or nothing case
-    match Option::<String>::deserialize(deserializer)? {
-        Some(name) => match name.len() {
-            0 => Ok(None),
-            1..=255 => Ok(Some(name)),
-            _ => Err(D::Error::custom("a room name cannot be more than 255 bytes")),
-        },
-        None => Ok(None),
+/// The name of a room.
+///
+/// It should not exceed 255 characters and should not be empty.
+#[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+#[serde(transparent)]
+#[allow(clippy::exhaustive_structs)]
```

Comment:
I'm pretty sure this can be removed, as the lint only triggers on structs with all-public fields in the first place.

Reply:
Yes, I think I'd originally just misread one of the clippy messages. I've removed it now.